internal/sync/syncer.go

```diff
@@ -343,8 +344,18 @@ func (s *Syncer) compareState(gitSync *v1alpha1.GitSync, targetObjs []*unstructu
 			IgnoreDifferences: OverrideIgnoreDiff{JSONPointers: []string{"/status"}}},
 	}
 
+	resourceOps, cleanup, err := s.getResourceOperations()
+	if err != nil {
+		log.Errorf("CompareAppState error getting resource operations: %s", err)
+	}
+	defer cleanup()
+
 	diffOpts := []diff.Option{
 		diff.WithLogr(*logger.New().WithValues("gitsync", fmt.Sprintf("%s/%s", gitSync.Namespace, gitSync.Name)).LogrLogger),
+		diff.WithServerSideDiff(true),
+		diff.WithServerSideDryRunner(diff.NewK8sServerSideDryRunner(resourceOps)),
+		diff.WithManager(common.SSAManager),
+		diff.WithGVKParser(clusterCache.GetGVKParser()),
```

Comment:
Why do we need this?

Reply:
without `WithGVKParser` serverSide diff was not working